app/src/main/java/de/danoeh/antennapod/fragment/preferences/SynchronizationPreferencesFragment.java

```diff
@@ -0,0 +1,294 @@
+
+import android.text.Spanned;
+import android.text.format.DateUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ImageView;
+import android.widget.ListAdapter;
+import android.widget.TextView;
+
+import com.google.android.material.snackbar.Snackbar;
+import com.nextcloud.android.sso.AccountImporter;
+import com.nextcloud.android.sso.exceptions.AccountImportCancelledException;
+import com.nextcloud.android.sso.exceptions.AndroidGetAccountsPermissionNotGranted;
+import com.nextcloud.android.sso.exceptions.NextcloudFilesAppNotInstalledException;
+import com.nextcloud.android.sso.helper.SingleAccountHelper;
+import com.nextcloud.android.sso.ui.UiExceptionManager;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.activity.PreferenceActivity;
+import de.danoeh.antennapod.core.event.SyncServiceEvent;
+import de.danoeh.antennapod.core.preferences.GpodnetPreferences;
+import de.danoeh.antennapod.core.sync.SyncService;
+import de.danoeh.antennapod.dialog.AuthenticationDialog;
+
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+
+import static de.danoeh.antennapod.core.sync.SyncService.SYNC_PROVIDER_CHOICE_GPODDER_NET;
+
+public class SynchronizationPreferencesFragment extends PreferenceFragmentCompat {
+    private static final String PREFERENCE_SYNCHRONIZATION_DESCRIPTION = "preference_synchronization_description";
+    private static final String PREFERENCE_LOGIN = "pref_synchronization_authenticate";
+    private static final String PREF_GPODNET_SETLOGIN_INFORMATION = "pref_gpodnet_setlogin_information";
+    private static final String PREFERENCE_SYNC = "pref_synchronization_sync";
+    private static final String PREFERENCE_FORCE_FULL_SYNC = "pref_synchronization_force_full_sync";
+    private static final String PREFERENCE_LOGOUT = "pref_synchronization_logout";
+    public static final String SYNC_PROVIDER_UNSET = "unset";
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        addPreferencesFromResource(R.xml.preferences_synchronization);
+        setupScreen();
+        updateScreen();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        ((PreferenceActivity) getActivity()).getSupportActionBar().setTitle(R.string.synchronization);
+        updateScreen();
+        EventBus.getDefault().register(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        EventBus.getDefault().unregister(this);
+        ((PreferenceActivity) getActivity()).getSupportActionBar().setSubtitle("");
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN, sticky = true)
+    public void syncStatusChanged(SyncServiceEvent event) {
+        updateScreen();
+        if (!GpodnetPreferences.loggedIn()) {
```

Comment:
The ActionBar subtitle should also be updated when syncing with your new service. Syncing (especially the first one) could take a pretty long time and users should see the current progress.